packages/core/scripts/admin-proposals/collateral.js

```diff
@@ -0,0 +1,327 @@
+// - Whitelist collateral on both (some on Ethereum, some on Polygon):
+//    - `HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/collateral.js --collateral 0xabc,0x123 --polygon 0xdef,`
+
+const hre = require("hardhat");
+require("dotenv").config();
+const { GasEstimator } = require("@uma/financial-templates-lib");
+const Web3 = require("web3");
+const winston = require("winston");
+const { parseUnits } = require("@ethersproject/units");
+const { interfaceName } = require("@uma/common");
+const { _getDecimals, _getContractAddressByName, _impersonateAccounts } = require("./utils");
+const argv = require("minimist")(process.argv.slice(), {
+  string: [
+    // comma-delimited list of final fees to set for whitelisted collateral.
+    "collateral",
+    // comma-delimited list of collateral addresses to whitelist. Required if --polygon is omitted.
+    "fee",
+    // comma-delimited list of Polygon collateral addresses to whitelist. Required if --collateral is omitted
+    "polygon",
+  ],
+  boolean: [
+    // set True if verifying, False for proposing.
+    "verify",
+  ],
+  default: { verify: false },
+});
+
+// Net ID returned by web3 when connected to a mainnet fork running on localhost.
+const HARDHAT_NET_ID = 31337;
+// Net ID that this script should simulate with.
+const PROD_NET_ID = 1;
+// Wallets we need to use to sign transactions.
+const REQUIRED_SIGNER_ADDRESSES = { deployer: "0x2bAaA41d155ad8a4126184950B31F50A1513cE25" };
+
+async function run() {
+  const { collateral, fee, polygon, verify } = argv;
+  const { getContract, web3, network, assert } = hre;
+
+  let netId = await web3.eth.net.getId();
+  if (netId === HARDHAT_NET_ID) {
+    console.log("🚸 Connected to a local node, attempting to impersonate accounts on forked network 🚸");
+    console.table(REQUIRED_SIGNER_ADDRESSES);
+    await _impersonateAccounts(network, REQUIRED_SIGNER_ADDRESSES);
+    console.log("🔐 Successfully impersonated accounts");
+  } else {
+    console.log("📛 Connected to a production node 📛");
+  }
+
+  // Contract ABI's
+  const ERC20 = getContract("ERC20");
+  const AddressWhitelist = getContract("AddressWhitelist");
+  const Store = getContract("Store");
+  const GovernorRootTunnel = getContract("GovernorRootTunnel");
+  const Governor = getContract("Governor");
+  const Finder = getContract("Finder");
+  const Voting = getContract("Voting");
+
+  // Parse comma-delimited CLI params into arrays
+  let collaterals;
+  let fees = fee.split(",");
+  let polygonCollaterals;
+  let crossChainWeb3;
+
+  // If polygon collateral is specified, initialize Governance relay infrastructure contracts
+  let polygon_netId;
+  let polygon_whitelist;
+  let polygon_store;
+  if (polygon) {
+    if (collateral) collaterals = collateral.split(",");
+    polygonCollaterals = polygon.split(",");
+    if (!process.env.CROSS_CHAIN_NODE_URL)
+      throw new Error("If --polygon is defined, you must set a CROSS_CHAIN_NODE_URL environment variable");
```

Comment:
nit: would it not make more sence to call the second provider `POLYGON_CHAIN_NODE_URL` as this is explicitly for a polygon whitelist & process?

Reply:
follow on question: now that you've set up this structure and worked with it a bit, how do you think it would best generalize to n number of other networks? if we had optimism support, what would you recommend doing? would we need to modify this script to have that additional logic within it? would you then want a separate ENV for the node URL?

more generally, is there a way we can restructure this to work with multiple networks without hardcoding names like this? granted, this will likely require more complexity and make the code hard to read but it is something to consider. its not worth it for this PR as we currently only have 2 networks supported but I wanted to start the discussion and see what you're thinking for this future support.

Reply:
> nit: would it not make more sense to call the second provider POLYGON_CHAIN_NODE_URL as this is explicitly for a polygon whitelist & process?

Sure, good call.

Reply:
> follow on question: now that you've set up this structure and worked with it a bit, how do you think it would best generalize to n number of other networks? if we had optimism support, what would you recommend doing? would we need to modify this script to have that additional logic within it? would you then want a separate ENV for the node URL?

I'd prefer to modify this script when Optimism support goes live. We may need to refactor this a bit and rename variables, but I don't want to over engineer any multichain support yet until we need it which might not be for another month.

Reply:
> more generally, is there a way we can restructure this to work with multiple networks without hardcoding names like this? granted, this will likely require more complexity and make the code hard to read but it is something to consider. its not worth it for this PR as we currently only have 2 networks supported but I wanted to start the discussion and see what you're thinking for this future support.

I think we would rename the `--polygon` flag to something else, but besides that it would work well. We could also get rid of the hardcoded `1.json` lookup but I do think we'll always be relaying governance from Ethereum mainnet to an L2.